### ISSUE TITLE
Threaded error

### ISSUE THREAD

**Issue by M-Zubair10** [QIN2DIM/hcaptcha-challenger#255]
Can you add timeout functionality, so in case it stuck anywhere, we'll have a backup solution
I did using timeout function of multiprocessing but it failed here due to asynchronous programming 
My function is:
```
def timeout_after(func, time_: float, args: tuple or list = (), kwargs: dict = {}):
    """
    Runs a function with time limit
    :param func: The function to run
    :param args: The functions args, given as tuple or list
    :param kwargs: The functions keywords, given as dict
    :param time_: The time limit in seconds
    :return: True if the function ended successfully. False if it was terminated.
    """
    pool = ThreadPool(processes=1)
    res = pool.apply_async(func, args=args, kwds=kwargs)
    try:
        return res.get(time_)
    except TimeoutError as e:
        return False
```

_Originally posted by @M-Zubair10 in https://github.com/QIN2DIM/hcaptcha-challenger/issues/252#issuecomment-1269725839_

**Comment by QIN2DIM** [comment]
You think this is multi-process asynchronous programming?

**Comment by M-Zubair10** [comment]
Yes

**Comment by QIN2DIM** [comment]
But you declare 'ThreadPool'

**Comment by M-Zubair10** [comment]
No, in your api I found async await keywords, I used threadpool so it might be interfering with async right?

**Comment by M-Zubair10** [comment]
![image](https://user-images.githubusercontent.com/47709300/194291859-73cd3246-a2aa-4a93-b916-6332a46e031a.png)
Thanks for speed efficient solution!

In 10 max retry limit,
here are the results, with 100 runs and average of 11sec

**Comment by QIN2DIM** [comment]
what？

**Comment by M-Zubair10** [comment]
You see I used ThreadPool for timeout request
But in your api I found async keywords
It means ThreadPool might be interfering with async event loop

**Comment by QIN2DIM** [comment]
Maybe you should research on google or stackoverflow, I don't know what you are talking about anymore🤨
